### PR TITLE
Fixes unintended synonyms of capitalized entities

### PR DESCRIPTION
**Pull request by twerkmeister**
Capitalized entities such as `China` are unintentionally added to the synonym dictionary. The dictionary then contains entries of the type lowercased_entity -> uppercased_entity. The additional problem is, that it overwrites intentional synonyms from the synonym lists. If the synonym lists contains China -> PRC then any training instance containing China will replace that entry with china -> China.  Also, I added a warning for when synonyms are being overwritten

Question: Was this previous behavior partionally intended and is being used? Could be that some people relied on the ner_synonyms component to capitalize things for them :bomb:

**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality

### CODE REVIEW

File: rasa_nlu/extractors/entity_synonyms.py [hunk 1]
```diff
@@ -85,8 +85,11 @@ def replace_synonyms(self, entities):
     def add_entities_if_synonyms(self, entity_a, entity_b):
         if entity_b is not None:
             original = entity_a if isinstance(entity_a, six.text_type) else six.text_type(entity_a)
-            original = original.lower()
             replacement = entity_b if isinstance(entity_b, six.text_type) else six.text_type(entity_b)
 
             if original != replacement:
+                original = original.lower()
+                if original in self.synonyms and self.synonyms[original] != replacement:
+                    warnings.warn("Overwriting existing synonym %s->%s with %s->%s" %
```
Comment (tmbo) [inline rasa_nlu/extractors/entity_synonyms.py]: Error message needs to be more expressive. Keep in mind, the user just wrote a training data file and now this warning pops up, what does he need to do?

**Comment by tmbo** [comment]
Please add a test and a note to the changelog.

**Comment by tmbo** [comment]
Reminder -  @twerkmeister  it would be great if you could address the open issues so we don't loose out on this cool contribution.

**Comment by twerkmeister** [comment]
Am back onto it!

**Comment by twerkmeister** [comment]
Updated

**Comment by twerkmeister** [comment]
I dug a bit into the test errors that make the build fail. Running them locally I figured out that the 404 return code comes from an InvalidProjectError being thrown during parsing. The exception contains the following message:
```
'/.../rasa_nlu/test_projects/test_project_mitie_sklearn/fallback/metadata.json'. [Errno 2] No such file or directory: '/.../rasa_nlu/test_projects/test_project_mitie_sklearn/fallback/metadata.json'
```
In this folder there are a bunch of models, but none called fallback. This does not seem to be related to my PR.

**Comment by twerkmeister** [comment]
Looked a bit more into it and it seems that the change with the glob in your last commit broke it (method _list_models_in_dir in project.py). The glob returns the full path, whereas the os.listdir only returned the basename. Not sure what this commit intended to change, so I didn't fix anything yet,  but I suppose it might have to do with the `_latest_project_model` method. That one is looking for the model_ prefix and timestamps

**Pull request merged**